spec/unit/container/auto_register_spec.rb

```diff
@@ -18,15 +18,38 @@ class Test::Container < Dry::Component::Container
     it { expect(Test::Container['bar.baz']).to be_an_instance_of(Bar::Baz) }
   end
 
+  context 'standard loader with a default namespace configured' do
+    before do
+      class Test::Container < Dry::Component::Container
+        configure do |config|
+          config.root = SPEC_ROOT.join('fixtures').realpath
+          config.namespace = 'namespaced'
+        end
+
+        load_paths!('components')
+        auto_register!('components/namespaced')
+      end
+    end
+
+    specify { expect(Test::Container['foo']).to be_a(Namespaced::Foo) }
+    specify { expect(Test::Container['bar']).to be_a(Namespaced::Bar) }
+  end
+
   context 'with a custom loader' do
     before do
       class Test::Loader < Dry::Component::Loader
-        def identifier
-          super.gsub('.', '-')
+        class Component < Dry::Component::Loader::Component
+          def identifier
+            super + ".yay"
+          end
+
+          def instance(*args)
+            constant.respond_to?(:call) ? constant : constant.new(*args)
+          end
         end
```

Comment:
this is *awesome*, feels like we could leverage that for our monitoring features